x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlConfig.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.parser;
+
+import org.elasticsearch.Build;
+
+class EsqlConfig {
+
+    // versioning information
+    boolean devVersion = Build.current().isSnapshot();
+
+    public void setDevVersion(boolean dev) {
+        this.devVersion = dev;
+    }
+
+    // not great because other grammar parser (Kibana) need the implement this method
+    boolean hasFeature(String featureName) {
+        return false;
+    }
```

Comment:
The point of this class is to hook the dev boolean flag and programatically set it for testing.
I've also added a potential hasFeature method to check if the underlying cluster supports certain command or not and disable it accordingly.

Reply:
Errr - do we expose feature flags through an API? How would Kibana determine if a feature flag is set on a given cluster?